python/pyarrow/tests/test_flight.py

```diff
@@ -1428,3 +1428,25 @@ def test_middleware_multi_header():
         for header, values in MultiHeaderClientMiddleware.EXPECTED.items():
             assert client_headers.get(header) == values
             assert headers.last_headers.get(header) == values
+
+
+@pytest.mark.requires_testing_data
+def test_generic_options():
+    """Test setting generic client options."""
+    certs = example_tls_certs()
+
+    with ConstantFlightServer(tls_certificates=certs["certificates"]) as s:
```

Comment:
Wouldn't it be simpler to test with `GRPC_ARG_MAX_RECEIVE_MESSAGE_LENGTH` as in C++?

Reply:
We do - there are two tests here, using the same properties as C++. I can split them if that's clearer.

Reply:
Ah, fine, I had misread.